deeplake/api/dataset.py

```diff
@@ -999,6 +1000,87 @@ def connect(
+            annotation_files (str, pathlib.Path, List[str]): Path to JSON annotation files in COCO format.
+            dest (str, pathlib.Path, Dataset):
+                - A Dataset or The full path to the dataset. Can be:
+                - a Deep Lake cloud path of the form ``hub://username/datasetname``. To write to Deep Lake cloud datasets, ensure that you are logged in to Deep Lake (use 'activeloop login' from command line)
+                - an s3 path of the form ``s3://bucketname/path/to/dataset``. Credentials are required in either the environment or passed to the creds argument.
+                - a local file system path of the form ``./path/to/dataset`` or ``~/path/to/dataset`` or ``path/to/dataset``.
+                - a memory path of the form ``mem://path/to/dataset`` which doesn't save the dataset but keeps it in memory instead. Should be used only for testing as it does not persist.
+            key_to_tensor_mapping (Optional[Dict]): A one-to-one mapping between COCO keys and Dataset tensor names.
+            file_to_group_mapping (Optional[Dict]): A one-to-one mapping between COCO annotation file names and Dataset group names.
+            ignore_one_group (bool): Skip creation of group in case of a single annotation file. Set to ``True`` by default.
+            ignore_keys (List[str]): A list of COCO keys to ignore.
+            image_settings (Optional[Dict]): A dictionary containing settings for the images tensor.
+            src_creds (Optional[Dict]): Credentials to access the source path. If not provided, will be inferred from the environment.
+            dest_creds (Optional[Dict]): A dictionary containing credentials used to access the destination path of the dataset.
+            inspect_limit (int): The maximum number of samples to inspect in the annotations json. Set to ``1000000`` by default.
+            progressbar (bool): Enables or disables ingestion progress bar. Set to ``True`` by default.
+            num_workers (int): The number of workers to use for ingestion. Set to ``0`` by default.
+            **dataset_kwargs: Any arguments passed here will be forwarded to the dataset creator function. See :func:`deeplake.dataset`.
+
+        Returns:
+            Dataset: The Dataset created from images and COCO annotations.
+
+        Raises:
+            IngestionError: If either ``key_to_tensor_mapping`` or ``file_to_group_mapping`` are not one-to-one.
+        """
+        dest = convert_pathlib_to_string_if_needed(dest)
+        images_directory = convert_pathlib_to_string_if_needed(images_directory)
+        annotation_files = (
+            [convert_pathlib_to_string_if_needed(f) for f in annotation_files]
+            if isinstance(annotation_files, list)
+            else convert_pathlib_to_string_if_needed(annotation_files)
+        )
+
+        if isinstance(dest, Dataset):
+            ds = dest
+        else:
+            dest = convert_pathlib_to_string_if_needed(dest)
```

Comment:
this is already done in deeplake.dataset, so probably not needed

Reply:
Oops, removed redundant path conversion